src/components/views/login/CaptchaForm.js

```diff
@@ -52,12 +52,24 @@ module.exports = React.createClass({
             this._onCaptchaLoaded();
         } else {
             console.log("Loading recaptcha script...");
-            var scriptTag = document.createElement('script');
             window.mx_on_recaptcha_loaded = () => {this._onCaptchaLoaded()};
-            scriptTag.setAttribute(
-                'src', global.location.protocol+"//www.google.com/recaptcha/api.js?onload=mx_on_recaptcha_loaded&render=explicit"
-            );
-            this.refs.recaptchaContainer.appendChild(scriptTag);
+            var protocol = global.location.protocol;
+            if (protocol === "file:") {
+                var warning = document.createElement('div');
+                // XXX: fix hardcoded app URL.  Better solutions include:
+                // * jumping straight to a hosted captcha page (but we don't support that yet)
+                // * embedding the captcha in an iframe (if that works)
+                // * using a better captcha lib
+                warning.innerHTML = "Robot check is currently unavailable on desktop - please sign up <a href='https://riot.im/app'>using a web browser</a>.";
```

Comment:
It's worth noting that this component is used whenever an operation demands interactiveauth with recaptcha, not just registration (see https://github.com/matrix-org/matrix-react-sdk/blob/69b277b28285eec9ab4c9f9fabcef194996db9bf/src/components/views/dialogs/InteractiveAuthDialog.js etc).

On the other hand, I don't think synapse actually requests a captcha for anything except registration currently, so I guess this wording is fine.